crates/re_renderer/src/allocator/gpu_readback_belt.rs

```diff
@@ -26,93 +36,134 @@ impl GpuReadbackBuffer {
         encoder: &mut wgpu::CommandEncoder,
         source: wgpu::ImageCopyTexture<'_>,
         copy_extents: glam::UVec2,
-    ) -> GpuReadbackBufferIdentifier {
-        let bytes_per_row =
-            TextureRowDataInfo::new(source.texture.format(), copy_extents.x).bytes_per_row_padded;
-
-        // Validate that stay within the slice (wgpu can't fully know our intention here, so we have to check).
-        // We go one step further and require the size to be exactly equal - there is no point in reading back more,
-        // as this would imply sniffing on unused memory.
-        debug_assert_eq!((bytes_per_row * copy_extents.y) as u64, self.size_in_bytes);
-        encoder.copy_texture_to_buffer(
-            source,
-            wgpu::ImageCopyBuffer {
-                buffer: &self.chunk_buffer,
-                layout: wgpu::ImageDataLayout {
-                    offset: self.byte_offset_in_chunk_buffer,
-                    bytes_per_row: NonZeroU32::new(bytes_per_row),
-                    rows_per_image: None,
-                },
-            },
-            wgpu::Extent3d {
-                width: copy_extents.x,
-                height: copy_extents.y,
-                depth_or_array_layers: 1,
-            },
-        );
-        self.identifier
+    ) {
+        self.read_multiple_texture2d(encoder, &[(source, copy_extents)]);
     }
 
-    /// Populates the buffer with data from a buffer.
-    pub fn read_buffer(
-        self,
+    /// Reads multiple textures into the same buffer.
+    ///
+    /// This is primarily useful if you need to make sure that data from all textures is available at the same time.
+    ///
+    /// Special care has to be taken to ensure that the buffer is large enough.
+    ///
+    /// ATTENTION: Keep in mind that internal offsets need to be a multiple of the texture block size!
+    /// While readback buffer starts are guaranteed to be aligned correctly, there might need to be extra padding needed between texture copies.
+    /// This method will add the required padding between the texture copies if necessary.
+    /// Panics if the buffer is too small.
+    pub fn read_multiple_texture2d(
```

Comment:
this became multiple for the sole purpose of handling depth textures later on. Struggled a lot with expressing the "I want several things to be done at once" and opted for the less user friendly way since I only have one usecase in mind right now and I think I know what I'm doing ;)